symbolic-common/src/byteview.rs

```diff
@@ -260,4 +283,28 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_mmap_fd_reuse() -> Result<(), std::io::Error> {
+        let mut tmp = NamedTempFile::new()?;
+        tmp.write_all(b"1234")?;
+
+        let view = ByteView::mmap_file(tmp.as_file())?;
+
+        // This deletes the file on disk.
+        let path = tmp.path().to_path_buf();
+        let mut file = tmp.into_file();
+        assert!(!path.exists());
+
+        // Ensure we can still read from the the file after mmapping and deleting it on disk.
+        let mut buf = Vec::new();
+        file.rewind()?;
+        file.read_to_end(&mut buf)?;
```

Comment:
So does mmap-ing keep the FD alive? Might be interesting to explicitly `drop(file)` after this, and see that the view still works (and holds on to the FD).

Reply:
Yes, mmap only uses the fd to figure out what to map, afterwards you can close it or do whatever you like to it.  I added the explicit drop to make it even more obvious

Reply:
> Yes,

That "yes" is not really true as the rest of the sentence explains 😄   mmap does not use the fd at all other than finding the right file to map.  so it doesn't keep the FD alive either.